components/pii_redaction/src/pii_redaction.py

```diff
@@ -0,0 +1,166 @@
+import json
+import random
+import string
+import ipaddress
+
+
+# List of random private IP addresses to use as replacements
+REPLACEMENTS_IP = {
+    "IPv4": [
+        "172.16.31.10",
+        "172.16.58.3",
+        "172.16.17.32",
+        "192.168.127.12",
+        "192.168.3.11",
+    ],
+    "IPv6": [
+        "fd00:c2b6:b24b:be67:2827:688d:e6a1:6a3b",
+        "fd00:a516:7c1b:17cd:6d81:2137:bd2a:2c5b",
+        "fc00:e968:6179::de52:7100",
+        "fc00:db20:35b:7399::5",
+        "fdf8:f53e:61e4::18",
+    ],
+}
+
+# providergs = ["google", "cloudfare", "alternate-dns", "quad9","open-dns", "comodo", "adguard"]
+POPULAR_DNS_SERVERS = [
+    "8.8.8.8",
+    "8.8.4.4",
+    "1.1.1.1",
+    "1.0.0.1",
+    "76.76.19.19",
+    "76.223.122.150",
+    "9.9.9.9",
+    "149.112.112.112",
+    "208.67.222.222",
+    "208.67.220.220",
+    "8.26.56.26",
+    "8.20.247.20",
+    "94.140.14.14",
+    "94.140.15.15",
+]
+
+
+def load_json(sample):
+    try:
+        return json.loads(sample)
+    except ValueError:
+        return []
+
+
+def random_replacements(n=10):
+    """Build dictionaries of random replacements for PII (key, email, IP address)
+
+    Emails: replace with one of n [random string of 5 characters + @example.com]
+    IP addresses: replace with one of n synthetic private IP addresses (IPv4 or IPv6)
+    Keys: replace with one of n [sequence of 32 random characters/digits]
+
+    TODO: add IPv6 and IPv4 separation
```

Comment:
This was taken from the BigCode project

Reply:
maybe not for this project but it would be interesting to have a way of working on updating components like this where the source code is taken from another repo. Right now it's just a snapshot

Reply:
Would be ideal if this was a library.